src/NUnitTestAdapter/NUnitTestAdapter.cs

```diff
@@ -203,9 +203,15 @@ private TestPackage CreateTestPackage(string assemblyName)
             {
                 package.Settings[PackageSettings.DomainUsage] = "Single";
             }
-
-            // Force truncation of string arguments to test cases
-            package.Settings[PackageSettings.DefaultTestNamePattern] = "{m}{a:40}";
+            if (Settings.DefaultTestNamePattern != null)
+            {
+                package.Settings[PackageSettings.DefaultTestNamePattern] = Settings.DefaultTestNamePattern;
+            }
+            else
+            {
+                // Force truncation of string arguments to test cases
+                package.Settings[PackageSettings.DefaultTestNamePattern] = "{m}{a:40}";
```

Comment:
Any desire to increase the default? Or do you think this would confuse people? A value of 80 might be worth considering for some values that are paths.